factory.js

```diff
@@ -106,50 +107,16 @@ Factory.prototype.init = function init(next) {
   const entry = this.data.entry;
   const base = this.base;
   const factory = this;
-  const pkg = path.join(base, 'package.json');
-  const wrhs = path.join(base, 'wrhs.toml');
 
-  //
-  // Read the package.json AND the wrhs.toml
-  //
-  debug(`Read package.json: ${ pkg } and wrhs.toml: ${ wrhs }`);
-  async.parallel([
-    function packJson(fn) {
-      fs.readFile(pkg, 'utf-8', function read(error, content) {
-        if (error) return void fn(error);
-
-        try {
-          factory.pkg = JSON.parse(content);
-          factory.entry = path.join(base, entry || factory.pkg.main);
-        } catch (err) {
-          return fn(err);
-        }
-
-        debug(`Set entry of factory to ${ factory.entry }`);
-        debug(`Parsed package.json content`, factory.pkg);
-        return fn();
-      });
-    },
-    function whrsCfg(fn) {
-      fs.readFile(wrhs, 'utf-8', function readme(error, content) {
-        if (error && error.code === 'ENOENT') return void fn();
-        if (error) return fn(error);
-
-        try {
-          factory.config = toml.parse(content);
-        } catch (err) {
-          return fn(err);
-        }
-
-        debug(`Parsed wrhs.toml content`, factory.config);
-        return fn();
-      });
-    }
-  ], function (err) {
-    if (err) return void next(err);
-
-    return void next();
-  });
+  extract(base)
+    .then(({ pkg, wrhs }) => {
+      debug(`Read configuration: ${ wrhs }`);
+      factory.pkg = pkg;
```

Comment:
Do we know why it needs the package.json? what is it doing with it? Is it just `browserify` that needs it?